tools/events-automation/README.md

```diff
@@ -1 +1,71 @@
+- date (date or datetime): Date of event in the locations local time (NOT the local time of where the program is being run).
+- url (string): Unique URL for event page details.
+- virtual (boolean): If event is online.
+- organizerName (string): Name of event organiser.
+- organizerUrl (string): URL for event organiser webpage.
+
+**Additional Variable(s) for Internal Use:**
+- duplicate (boolean): Flag for potential event duplicate based on evaluation during event sink. 
+
+**Code Sample for Creating an Event:**
+```py
+Event(name="Test Event", location="Melbourne, VIC, AU", date=date.today(), url="website3.com", virtual=True, organizerName="Test Organizer", organizerUrl="testorg.com")
+```
+
+### Event Sink:
+**Within Scope**:
+- The event sink will take a list of event objects (see `test_events.py` for example data), format the date and location data (if not done already), filter out events that are outside of the pre-determined 'date window' for the current TWiR issue then sort the events by date and then location alphabetically. After this process the list is then split via virtual or continent, and any potential duplicate events within the list are flagged (through comparison of event data). Finally, the event sink will output the details of the finalised list of events in a pre-determined markdown format, complete with virtual/continent headers.
+- Note that potential duplicate events will be flagged with a `** NOTE POTENTIAL DUPLICATE: **` warning immediately preceding the event information.
+
+**Out of Scope**:
+- The purpose of the event sink is to cross-reference and curate data from various sources. It shouldn't be responsible for gathering or adding required fields of data into the Event class. Any edge cases should be managed by the event sources.
+
+### How to Add a New Event Source Module:
+- To write a new event source module, it must be written in python. Event sources should be a function that passes no parameters and returns a list of events with required variables detailed above. If the event source has additional requirements, that should be added to `requirements.txt`. The event source should detail any specific run instructions in it's own documentation. Look at `test_events.py` for bare-minimum event list output.
+- To run a new event source, import the new module from new event source and add the function call to `event_list`.
+
+### Requirements to run this code:
+- Event sink requires Python installation.
+- For specific module requirements: `pip install -r requirements.txt`
+
+### How to use this code:
+- In order to use the Event Sink code please ensure that all requirements listed above have been met/installed on your device.
+- Check that all Event Source module function calls are included in `event_list` (function calls should concatenate into a single list of event objects).
```

Comment:
Is this suggesting that the current assignment of the output of `get_test_events` should be replaced with the actual input data?
I might replace this readme with TODO inline with the code @ main.py  at the line where the actual input data function call needs to be swapped in.